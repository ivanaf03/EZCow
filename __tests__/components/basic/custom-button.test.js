import React from "react";

import { render, fireEvent } from "@testing-library/react-native";

import CustomButton from "../../../components/basic/custom-button";

describe("CustomButton", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should render correctly", () => {
    const tree = render(<CustomButton text="Test" onPress={() => {}} />);
    expect(tree).toMatchSnapshot();
  });

  it("should call onPress when pressed", () => {
    const onPress = jest.fn();
    const tree = render(<CustomButton text="Test" onPress={onPress} />);
    fireEvent.press(tree.getByText("Test"));
    expect(onPress).toHaveBeenCalledTimes(1);
  });
});
