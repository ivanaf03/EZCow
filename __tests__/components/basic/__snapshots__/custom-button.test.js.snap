// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton should render correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "backgroundColor": "#272727",
      },
      {
        "borderRightWidth": 8,
      },
      {
        "borderLeftWidth": 8,
      },
      {
        "borderBottomColor": "#2254ba",
        "borderLeftColor": "#2254ba",
        "borderRightColor": "#2254ba",
        "borderTopColor": "#2254ba",
      },
      {
        "borderBottomLeftRadius": 24,
        "borderBottomRightRadius": 24,
        "borderTopLeftRadius": 24,
        "borderTopRightRadius": 24,
      },
      {
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
      {
        "marginBottom": 8,
        "marginLeft": 8,
        "marginRight": 8,
        "marginTop": 8,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#ffffff",
        },
        {
          "fontSize": 18,
          "lineHeight": 28,
        },
        {
          "textAlign": "center",
        },
        {
          "fontFamily": "Nunito-Medium",
        },
        {
          "marginLeft": 8,
          "marginRight": 8,
        },
      ]
    }
  >
    Test
  </Text>
</View>
`;
